### ISSUE TITLE
`default-param` transform shouldn't apply between parameters

### ISSUE THREAD

**Issue by josephfrazier** [lebab/lebab#238]
For example, this code:

```js
function f (a, b) {
  a = a || b
  console.log(a)
}
f(undefined, 2)
```

is perfectly valid and prints `2`. After the `default-param` transform, it becomes:

```js
function f(a = b, b) {
  console.log(a)
}
f(undefined, 2)
```

which throws `ReferenceError: b is not defined`. Demo: https://runkit.com/josephfrazier/runkit-npm-lebab-default-param-between-parameters/1.0.0

---

I think the `default-param` transform should ensure that if the default value is a variable reference, then the variable must be in scope outside the function.

**Comment by nene** [comment]
Thanks for the fix.

I tweaked the code a bit and merged it in.

It doesn't cover all cases like:

```js
function f(a, b) {
    a = a || b/2;
    console.log(a);
}
```

but I guess it covers the most common case.

**Comment by nene** [comment]
I think I better reopen this issue, as it's not fully resolved.

**Comment by nene** [comment]
Fixed in 3.0.2
